vars/buildFailureMessage.groovy

```diff
@@ -0,0 +1,17 @@
+@NonCPS
+def call(){
+    def ERROR_STRING = "ERROR"
+    def message = ""
+    String delimiter = ","
+    def performance_log = currentBuild.rawBuild.getLog(Integer.MAX_VALUE).join('\n')
+    performance_log.eachLine() { line ->
+        if (line.matches(".*$ERROR_STRING.*")) {
+            message+=(line+delimiter)
+        }
+    }
+    message=message.replaceAll(",", "\n")
+    if(message.isEmpty()){
+        message="Build failed"
+    }
```

Comment:
```suggestion
    if(message.isEmpty()){
        message="Build failed"
    } else {
        message=message.replaceAll(",", "\n")
    }
```

Reply:
We want this to message.replaceAll(",", "\n") even-though message.isEmpty() is true as want to separate the Error messages one per line
Example slack output
```
MESSAGE=2022-03-24 15:30:20 ERROR    Error building job-scheduler
 retry with: ./build.sh manifests/2.0.0/opensearch-2.0.0.yml --component job-scheduler --snapshot
2022-03-24 15:31:41 ERROR    Error building job-scheduler
 retry with: ./build.sh manifests/2.0.0/opensearch-2.0.0.yml --component job-scheduler --snapshot
```

Reply:
if message is an empty string, what will `.replaceAll` replace?